src/main/java/az/cybernet/internship/dictionary/service/DictionaryServiceHandler.java

```
@@ -0,0 +1,72 @@
+        var dictionary = fetchDictionaryIfExists(id);
+        return new DictionaryResponse(
+                dictionary.getId(),
+                dictionary.getValue(),
+                dictionary.getDescription(),
+                dictionary.getDeletedAt()
+        );
+    }
+
+    public void saveDictionary(CreateDictionaryRequest request) {
+        dictionaryRepository.save(DictionaryEntity.builder()
+                .value(request.getValue())
+                .description(request.getDescription())
+                .status(DictionaryStatus.ACTIVE)
+                .build());
+    }
+
+    public void updateDescription(Long id, UpdateDescriptionRequest request) {
+        var dict = dictionaryRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.BAD_REQUEST));
+        dict.setDescription(request.getDescription());
+        dictionaryRepository.save(dict);
+    }
+
+    public void deleteDictionary(Long id) {
+        var dict = dictionaryRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+        dict.setStatus(DictionaryStatus.DELETED);
+        dict.setDeletedAt(LocalDateTime.now());
+        dictionaryRepository.save(dict);
+    }
+
+    public void restoreDictionary(Long id) {
+        var dict = dictionaryRepository.findById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+
+        if (dict.getStatus() == DictionaryStatus.ACTIVE) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Dictionary is already active");
+        }
+
+        dict.setStatus(DictionaryStatus.ACTIVE);
+        dict.setDeletedAt(null);
+        dictionaryRepository.save(dict);
+    }
+
+    private DictionaryEntity fetchDictionaryIfExists(Long id) {
+        return dictionaryRepository.findByIdAndStatusNot(id, DictionaryStatus.DELETED)
+                .orElseThrow(RuntimeException::new);
```

silent failure - .orElseThrow(RuntimeException::new);
exception aydın deyil. ümumiyyətlə logging mexanizmi mövcud deyil. troubleshooting prosesi çətinləşəcək